libkbfs/md_journal.go

```diff
@@ -681,13 +681,14 @@ func (j *mdJournal) convertToBranch(
 		prevID = newID
 
 		// If possible, replace the old RMD in the cache.  If it's not
-		// already in the cache, don't bother adding it, as that will
-		// just evict something incorrectly.  TODO: Don't replace the
-		// MD until we know for sure that the branch conversion
-		// succeeds.
+		// already in the cache, or if it's been replaced by the REAL
+		// commit from the master branch due to a race, don't bother
+		// adding it, as that will just evict something incorrectly.
```

Comment:
Much worse can happen if it's been replaced by the real commit, right? Perhaps reword as:

```
in the cache, don't bother adding it ...incorrectly. If it's been replaced by the REAL commit ... race, also don't clobber it.
```